src/Microsoft.Identity.Web/ClientAssertionDescription.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Description of a client assertion.
+    /// </summary>
+    public class ClientAssertionDescription
+    {
+        /// <summary>
+        /// Constructor of a ClientAssertionDescription.
+        /// </summary>
+        /// <param name="clientAssertionProvider">delegate providing the client assertion
+        /// when it is necessary.</param>
+        public ClientAssertionDescription(Func<CancellationToken, Task<ClientAssertion>> clientAssertionProvider)
+        {
+            ClientAssertionProvider = clientAssertionProvider;
+        }
+
+        /// <summary>
+        /// delegate to get the client assertion.
+        /// </summary>
+        protected Func<CancellationToken, Task<ClientAssertion>> ClientAssertionProvider { get; private set; }
+
+        /// <summary>
+        /// Client assertion.
+        /// </summary>
+        private ClientAssertion? _clientAssertion;
+
+        /// <summary>
+        /// Get the signed assertion (and refreshes it if needed).
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The signed assertion.</returns>
+        public async Task<string> GetSignedAssertion(CancellationToken cancellationToken)
+        {
+            if (_clientAssertion == null || (Expiry != null && DateTime.Now > Expiry))
```

Comment:
>DateTime.Now [](http://example.com/codeflow?start=63&length=12)

should this be UTC?

Reply:
Thanks, @jennyf19. I'll use `DateTimeOffset.Now` instead, now that I understand what this is :-)